lib/inspec/dependencies/requirement.rb

```diff
@@ -24,64 +47,81 @@ def matches_spec?(spec)
+        h['content_hash'] = content_hash
+      end
+      h
+    end
+
+    def lock_deps(dep_array)
+      @dependencies = dep_array
+    end
+
+    def is_vendored?
+      @vendor_index.exists?(@name, source_url)
+    end
+
+    def content_hash
+      @content_hash ||= begin
+                          archive_path = @vendor_index.archive_entry_for(@name, source_url)
+                          fail "No vendored archive path for #{self}, cannot take content hash" if archive_path.nil?
+                          Digest::SHA256.hexdigest File.read(archive_path)
+                        end
+    end
+
     def source_url
-      case source_type
-      when :local_path
-        "file://#{File.expand_path(opts[:path])}"
-      when :url
-        @opts[:url]
+      if opts[:path]
+        "file://#{opts[:path]}"
+      elsif opts[:url]
+        opts[:url]
       end
     end
 
     def local_path
-      @local_path ||= case source_type
-                      when :local_path
-                        File.expand_path(opts[:path], @cwd)
+      @local_path ||= if fetcher.class == Fetchers::Local
+                        File.expand_path(fetcher.target, @cwd)
                       else
                         @vendor_index.prefered_entry_for(@name, source_url)
                       end
     end
 
-    def source_type
-      @source_type ||= if @opts[:path]
-                         :local_path
-                       elsif opts[:url]
-                         :url
-                       else
-                         fail "Cannot determine source type from #{opts}"
-                       end
-    end
-
-    def fetcher_class
-      @fetcher_class ||= case source_type
-                         when :local_path
-                           Fetchers::Local
-                         when :url
-                           Fetchers::Url
-                         else
-                           fail "No known fetcher for dependency #{name} with source_type #{source_type}"
-                         end
-
-      fail "No fetcher for #{name} (options: #{opts})" if @fetcher_class.nil?
-      @fetcher_class
+    def fetcher
+      @fetcher ||= Inspec::Fetcher.resolve(source_url)
+      fail "No fetcher for #{name} (options: #{opts})" if @fetcher.nil?
+      @fetcher
     end
 
     def pull
-      case source_type
-      when :local_path
+      # TODO(ssd): Dispatch on the class here is gross.  Seems like
+      # Fetcher is missing an API we want.
+      if fetcher.class == Fetchers::Local || @vendor_index.exists?(@name, source_url)
         local_path
       else
-        if @vendor_index.exists?(@name, source_url)
-          local_path
-        else
-          archive = fetcher_class.download_archive(source_url)
-          @vendor_index.add(@name, source_url, archive.path)
-        end
+        @vendor_index.add(@name, source_url, fetcher.archive.path)
+      end
+    end
+
+    def dependencies
+      return @dependencies unless @dependencies.nil?
```

Comment:
Ah, this is mostly because this function used to be a lot longer so it would have been `@dependencies ||= begin` where the begin block was really big.  I don't think dependencies can be nil after this because 118 either will return a map or raise some NoMethodError.  I suppose it can be nil if someone calls lock_deps(nil), but in that case I feel like you should know what you are doing.

Anyway, I can move this to our more typical style.